### PR TITLE
test-network: fixes for test_activation_policy

### LINKED ISSUE #22267: test-network: test_activation_policy is racy
```python
        start_networkd()

        always = test.startswith('always')
        if test == 'manual':
            initial_up = 'UP' in check_output('ip link show test1')    # <--- This is racy
        else:
            initial_up = not test.endswith('down') # note: default is up
        expect_up = initial_up
        next_up = not expect_up
```
See https://jenkins-systemd.apps.ocp.ci.centos.org/job/upstream-vagrant-archlinux/9726/artifact//systemd-centos-ci/index.html

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #22267.

### CODE REVIEW

File: test/test-network/systemd-networkd-tests.py [hunk 1]
```diff
@@ -3101,6 +3101,8 @@ def _test_activation_policy(self, test):
                 check_output('ip link set dev test1 down')
             expect_up = initial_up if always else next_up
             next_up = not next_up
+            if always:
+                time.sleep(1)
```
Comment (anitazha) [inline test/test-network/systemd-networkd-tests.py:3105]: Just wondering if there is there a more reliable way to tell if the state propagates to networkd? Sometimes the hardcoded sleeps aren't enough on a slow CI.

**Review by anitazha — APPROVED** [review]
LGTM one question but not a blocker

File: test/test-network/systemd-networkd-tests.py (hunk 1 above)
Comment (yuwata) [reply test/test-network/systemd-networkd-tests.py:3105]: Unfortunately, I do not have any good idea... Potentially, the following should work: invoke `journalctl --follow` before `ip link set`, and monitor the log until it contains `"ActivationPolicy is \"always-on\", forcing link up."`. But that sounds quite overkill, and it is had for me as I am not good at writing Python...

I'd like to keep this as is until the hardcoded sleep causes CI failures many times.

**Comment by yuwata** [comment]
Let's merge this, as @anitazha provides positive review, and find a better way later if the hadcoded sleep does not work fine.

**Pull request merged**